frontend/src/components/starry-map/StarryMap.vue

```diff
@@ -0,0 +1,275 @@
+          {{ star.text }}
+        </div>
+      </div>
+    </div>
+    <div v-if="selectedStar" class="info-box">
+      <h3>{{ selectedStar.name }}</h3>
+      <p>Type: {{ selectedStar.type }}</p>
+      <p>Users: {{ selectedStar.users.join(', ') }}</p>
+      <p>Description: {{ selectedStar.description }}</p>
+      <button @click="selectedStar = null">Close</button>
+    </div>
+    <div class="controls">
+      <div class="position-info">{{ positionInfo }}</div>
+      <div class="zoom-control">
+        <button v-for="zoomLevel in zoomLevels" :key="zoomLevel.value" 
+                class="zoom-button" 
+                @click="setZoom(zoomLevel.value)">
+          {{ zoomLevel.label }}
+        </button>
+      </div>
+    </div>
+  </template>
+  
+  <script lang="ts">
+  import { defineComponent, ref, computed, onMounted, onBeforeUnmount } from 'vue';
+  
+  interface Star {
+    id: number;
+    left: string;
+    top: string;
+    pulsing: boolean;
+    text: string;
+    name: string;
+    type: string;
+    users: string[];
+    description: string;
+  }
+  
+  interface Layer {
+    src: string;
+  }
+  
+  interface ZoomLevel {
+    label: string;
+    value: number;
+  }
+  
+  export default defineComponent({
+    name: 'StarryMap',
+    setup() {
+      const layers = ref<Layer[]>([
+        { src: 'src/assets/img/star_bg3.png' },
+        { src: 'src/assets/img/star_bg2.png' },
+        { src: 'src/assets/img/star_bg1.png' },
+      ]);
+      const stars = ref<Star[]>([]);
+      const viewportX = ref(0);
+      const viewportY = ref(0);
+      const scale = ref(1);
+      const isDragging = ref(false);
+      const startX = ref(0);
+      const startY = ref(0);
+      const selectedStar = ref<Star | null>(null);
+      const zoomLevels: ZoomLevel[] = [
+        { label: '--', value: 0.5 },
+        { label: '-', value: 0.8 },
+        { label: 'o', value: 1 },
+        { label: '+', value: 1.3 },
+        { label: '++', value: 1.6 },
+      ];
+  
+      const positionInfo = computed(() => {
+        return `Position: (${Math.round(viewportX.value)}, ${Math.round(viewportY.value)}) Zoom: ${scale.value.toFixed(2)}`;
+      });
+  
+      const generateStars = () => {
+        const starTypes = ['User', 'Chat Room', 'Forum', 'Game Room', 'Workshop'];
+        stars.value = Array.from({ length: 200 }, (_, i) => ({
+          id: i + 1,
+          left: `${Math.random() * 100}%`,
+          top: `${Math.random() * 100}%`,
+          pulsing: Math.random() < 0.1,
+          text: starTypes[Math.floor(Math.random() * starTypes.length)].charAt(0),
+          name: `${starTypes[Math.floor(Math.random() * starTypes.length)]} ${i + 1}`,
+          type: starTypes[Math.floor(Math.random() * starTypes.length)],
+          users: Array.from({ length: Math.floor(Math.random() * 5) + 1 }, () => `User${Math.floor(Math.random() * 1000)}`),
+          description: `This is a virtual space. More information can be added here.`
+        }));
```

Comment:
```suggestion
        stars.value = Array.from({ length: 200 }, (_, i) => {
          const id = id: i + 1
          const description = 'This is a virtual space. More information can be added here.'
          const left = `${Math.random() * 100}%`
          const top = `${Math.random() * 100}%`
          const pulsing = Math.random() < 0.1
          const randomIndex = Math.floor(Math.random() * starTypes.length)
          const type = starTypes[randomIndex]
          const text = type.charAt(0)
          const name = `${type} ${i + 1}`
          const users = Array.from({ length: Math.floor(Math.random() * 5) + 1 }, () => `User${Math.floor(Math.random() * 1000)}`),
          return { id, left, top, pulsing, type, text, name users }
        });
```

This ensures that we use the same `randomIndex` instead of re-evaluating.